lib/internal/fs/promises.js

```diff
@@ -274,7 +278,21 @@ async function fsCall(fn, handle, ...args) {
 }
 
 async function writeFileHandle(filehandle, data, signal) {
-  // `data` could be any kind of typed array.
+  if (signal?.aborted) {
+    throw lazyDOMException('The operation was aborted', 'AbortError');
+  }
+  if (isCustomIterable(data)) {
+    for await (const buf of data) {
+      if (signal?.aborted) {
+        throw lazyDOMException('The operation was aborted', 'AbortError');
+      }
+      await filehandle.write(buf);
```

Comment:
This should be equivalent and have less overhead.
```suggestion
      await write(filehandle, buf);
```
I think we should document that the `encoding` option is ignored when passing an iterable – or ideally we should not ignore it.

Reply:
@aduh95 
thanks, I fixed it.